activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -206,9 +219,13 @@ def lease
         @owner = Thread.current
       end
 
+      def schema_cache
+        @pool.get_schema_cache(self)
+      end
+
       def schema_cache=(cache)
         cache.connection = self
-        @schema_cache = cache
+        @pool.set_schema_cache(cache)
```

Comment:
We should find places calling `schema_cache=` on the connection and remove them.  Since the connection just asks the pool for the schema cache, there's no reason to set it on the connection AFAIK.